src/Discovery/test/ClientBase.Test/NoOpDiscoveryClientTest.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Steeltoe.Discovery.Client.SimpleClients;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Steeltoe.Discovery.ClientBase.Test
+{
+    public class NoOpDiscoveryClientTest
+    {
+        [Fact]
+        public void NoOpDiscoveryClient_LooksForOtherClients()
+        {
+            // Arrange
+            var appsettings = new Dictionary<string, string>()
+            {
+                { "spring:application:name", "myName" },
+                { "Consul:somekey", "somevalue" },
+                { "Eureka:somekey", "somevalue" },
+                { "spring:cloud:kubernetes:discovery:somekey", "somevalue" },
+                { "DiscoveryClients:TestClient", "TestClientConfigPath" },
+                { "TestClientConfigPath:Uri", "http://someserver:1234" }
+            };
+            var config = new ConfigurationBuilder().AddInMemoryCollection(appsettings).Build();
+            var logger = new Mock<ILogger<NoOpDiscoveryClient>>();
```

Comment:
This works and is fine as is. I did run into a few issues with logging, testing logging and logging while debugging tests and we have solved this in several ways in the code base. perhaps we can standardize on this in the future if it works for us : https://github.com/roryprimrose/Divergic.Logging.Xunit